src/test/fio/fio_ceph_objectstore.cc

```diff
@@ -20,6 +20,10 @@
 #include "include/random.h"
 #include "common/perf_counters.h"
 
+#ifndef typeof
+#define typeof __typeof__
+#endif
+
```

Comment:
not sure why this is necessary. src/test/fio/CMakeLists.txt has a workaround for this:
```cmake
# fio headers use typeof(), which requires c++11 extensions
set_target_properties(fio_ceph_objectstore PROPERTIES
  CXX_EXTENSIONS ON
  COMPILE_FLAGS "${FIO_CFLAGS}")
```

Reply:
This probably won't be it in the final version of this PR. However, I was planning on submitting it alone as the fio_ceph_objectstore.cc wouldn't build in my environment. I haven't tried it since though.

Reply:
i also ran into this issue while compiling `libfio_ceph_objectstore.so` with GCC 8.2, hence https://github.com/axboe/fio/pull/628 , which renders both workaround not necessary anymore.